x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/RealmSettings.java

```diff
@@ -181,6 +184,7 @@ private static void validateRealm(String name, String type, Settings settings, S
         settingSet.add(TYPE_SETTING);
         settingSet.add(ENABLED_SETTING);
         settingSet.add(ORDER_SETTING);
+        settingSet.add(LOOKUP_REALMS_SETTING);
```

Comment:
I'm of two minds here.
I understand the desire to make this a common setting with consistent behaviour across all realms, but I think it ultimately causes problems.
This implies that it's OK to set `lookup_realms` on any realm, but each realm needs to implement their own support for lookup realms, so it will be possible to create configurations that appear to work, but ultimately do nothing. I think that's a mistake.

My personal preference would be to implement lookup support as a standalone utility class and then hook that into each realm as we update them to support lookups.